src/appleseed/renderer/kernel/rendering/final/adaptivetilerenderer.cpp

```diff
@@ -164,10 +164,12 @@ namespace
         const float rcp_second_weight = second_weight == 0.0f ? 0.0f : 1.0f / second_weight;
 
         // Get colors and assign weights.
-        Color4f main_color(main[0], main[1], main[2], main[3]);
+        Color3f main_color(main[0], main[1], main[2]);
+        main_color = linear_rgb_to_srgb(main_color);
```

Comment:
Wait a minute, shouldn't the color space conversion happen _after_ the division by the weight? I'm pretty sure it should.

Reply:
Nice catch, yes!